daemon/commit.go

```diff
@@ -126,9 +126,9 @@ func (daemon *Daemon) Commit(name string, c *backend.ContainerCommitConfig) (str
 		return "", err
 	}
 
-	// It is not possible to commit a running container on Windows
-	if runtime.GOOS == "windows" && container.IsRunning() {
-		return "", fmt.Errorf("Windows does not support commit of a running container")
+	// It is not possible to commit a running container on Windows and on Solaris.
+	if (runtime.GOOS == "windows" || runtime.GOOS == "solaris") && container.IsRunning() {
```

Comment:
Honestly I think that would be overkill for one feature.. A multi-platform check such as this is done only in this file (all others are in test files and not feature specific). If such checks show up for multiple features in the future I'd be happy to implement that. For now I think its most readable this way.